dom/observable/tentative/observable-ctor.any.js

```diff
@@ -0,0 +1,706 @@
+}, "Errors pushed throw the subscriber that are not handled by the subscription are reported");
+
+test(() => {
+  const error = new Error("error");
+  let errorReported = false;
+
+  self.addEventListener(
+    "error",
+    (e) => {
+      assert_equals(e.message, "Uncaught (in observable) error");
+      assert_equals(e.filename, location.href);
+      assert_greater_than(e.lineno, 0);
+      assert_greater_than(e.colno, 0);
+      assert_equals(e.error, error);
+      errorReported = true;
+    },
+    { once: true }
+  );
+
+  const source = new Observable((subscriber) => {
+    throw error;
+  });
+
+  // No error handler provided.
+  source.subscribe({
+    next: () => assert_unreached("next should not be called"),
+    complete: () => assert_unreached("complete should not be called"),
+  });
+
+  assert_true(errorReported);
+}, "Errors thrown in the initializer that are not handled by the subscription are reported");
+
+test(() => {
+  const error = new Error("error");
+  const results = [];
+  let errorReported = false;
+
+  self.addEventListener(
+    "error",
+    (e) => {
+      assert_equals(e.message, "Uncaught (in observable) error");
+      assert_equals(e.filename, location.href);
+      assert_greater_than(e.lineno, 0);
+      assert_greater_than(e.colno, 0);
+      assert_equals(e.error, error);
+      errorReported = true;
+    },
+    { once: true }
+  );
+
+  const source = new Observable((subscriber) => {
+    subscriber.next(1);
+    subscriber.next(2);
+    subscriber.complete();
+    subscriber.error(error);
+  });
+
+  source.subscribe({
+    next: (x) => results.push(x),
+    error: () => assert_unreached("error should not be called"),
+    complete: () => results.push("complete"),
+  });
+
+  assert_array_equals(
+    results,
+    [1, 2, "complete"],
+    "should emit values synchronously, but not error values after complete"
+  );
+
+  assert_true(errorReported);
+}, "subscription reports errors that are pushed after subscriber is closed by completion");
+
+test(() => {
+  const error = new Error("error");
+  const results = [];
+  let errorReported = false;
+
+  self.addEventListener(
+    "error",
+    (e) => {
+      assert_equals(e.message, "Uncaught (in observable) error");
+      assert_equals(e.filename, location.href);
+      assert_greater_than(e.lineno, 0);
+      assert_greater_than(e.colno, 0);
+      assert_equals(e.error, error);
+      errorReported = true;
+    },
+    { once: true }
+  );
+
+  const source = new Observable((subscriber) => {
+    subscriber.next(1);
+    subscriber.next(2);
+    subscriber.complete();
+    throw error;
+  });
+
+  source.subscribe({
+    next: (x) => results.push(x),
+    error: () => assert_unreached("error should not be called"),
+    complete: () => results.push("complete"),
+  });
+
+  assert_array_equals(
+    results,
+    [1, 2, "complete"],
+    "should emit values synchronously, but not error after complete"
+  );
+
+  assert_true(errorReported);
```

Comment:
Should the error reporting handler be called _twice_? And if so, should we assert that one way or another?